### ISSUE TITLE
Ideas for Node-design

### ISSUE THREAD

**Issue by grindhold** [benwaffle/pulse-flow#3]
Here are some Ideas how to design the nodes in a way that the capabilities of all five tabs of pavucontrol may be merged into a single graph-based view without overloading the UI.

This is the design for a node that represents a program that generates an audio stream:
![Output Node](https://i.imgur.com/Us1b6Kp.png)
Recording programs may be modeled simultaneously.

This is a design for a Node that represents an Audio device:
![Device Node](https://i.imgur.com/R2tWADr.png)

This combines the functionalities of the tabs "Output Devices", "Input  Devices" and "Configuration". 
It combines the tab-entries for each device in one coherent ui-element.

This can be done via a custom node-renderer that renders one child for each entry as an input / output device.

Correction: I drew this without any references in public transport, which lead me to the wrong impression that a user could select the port of an output device on an per-application basis, what has proven to be incorrect after looking it up. please imagine the "headphone/rear out"-ports as one dock. and one more port-selection-dropdownbox in the children.

Another note: You requested the multiple connection sink feature in order to be able to connect more that one audiosource to a device. This will prove unhandy in practice, because the connections made to these sinks behave like a stack. You can only ever undo the topmost connection. This is why i introduced the before_linking signal. With this signal you cann hijack the connection-making-process, create a new dock per connection on-the-fly and connect the new audio-source to the newly created dock instead of making the sink a multisink.

Let me know what you think of those ideas.

**Comment by benwaffle** [comment]
>You can only ever undo the topmost connection.

Yeah I noticed there's no way to control what you disconnect. Multiple docks sounds reasonable
